common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -206,6 +208,23 @@ def handle_ajax(self, dispatch, data):  # TODO: bounds checking
                 self.position = 1
             return json.dumps({'success': True})
 
+        if dispatch == 'get_completion':
+            if not self.is_user_authenticated(context=data):
```

Comment:
I think `data` will look like this `{'usage_key': 'some-usage-key'}` - that's all that's being passed in from the `Sequence.update_completion()` javascript function above.  That is, it won't contain the `user_authenticated` key that `is_user_authenticated()` looks for, and will default to True. I think that means every user, authenticated or not, can get back this completion data.  I'm not sure if that's a big deal.

Reply:
Yeah, this is a useless check in this function. I just removed.